### PR TITLE
x509_crl_info: move main code to module_utils to allow easier implementation of diff mode

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Subset of #150.

##### ISSUE TYPE
- Refactoring Pull Request

##### COMPONENT NAME
x509_crl_info

### CODE REVIEW

File: plugins/module_utils/crypto/module_backends/crl_info.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self.content = content
+
+    def get_info(self):
+        self.crl_pem = identify_pem_format(self.content)
+        try:
+            if self.crl_pem:
+                self.crl = x509.load_pem_x509_crl(self.content, default_backend())
+            else:
+                self.crl = x509.load_der_x509_crl(self.content, default_backend())
+        except Exception as e:
```
Comment (Ajpantuso) [inline plugins/module_utils/crypto/module_backends/crl_info.py:61]:
```suggestion
        except ValueError as e:
```
See [here](https://github.com/pyca/cryptography/blob/1b922ed1dee0cd7e165a868639ce6d0869c8b2f5/src/cryptography/hazmat/backends/openssl/backend.py#L1346-L1370)

File: plugins/modules/x509_crl_info.py [hunk 2]
```diff
@@ -164,99 +161,9 @@
     identify_pem_format,
 )
```
Comment (Ajpantuso) [inline plugins/modules/x509_crl_info.py:163]: The imports between `OpenSSLObjectError` and `identify_pem_format` are stale now

File: plugins/modules/x509_crl_info.py [hunk 3]
```diff
@@ -274,13 +181,22 @@ def main():
...
     )
 
-    if not CRYPTOGRAPHY_FOUND:
-        module.fail_json(msg=missing_required_lib('cryptography >= {0}'.format(MINIMAL_CRYPTOGRAPHY_VERSION)),
-                         exception=CRYPTOGRAPHY_IMP_ERR)
+    if module.params['content'] is None:
+        try:
+            with open(module.params['path'], 'rb') as f:
+                data = f.read()
+        except Exception as e:
```
Comment (Ajpantuso) [inline plugins/modules/x509_crl_info.py:188]:
```suggestion
        except (IOError, OSError) as e:
```

File: plugins/modules/x509_crl_info.py [hunk 4]
```diff
@@ -274,13 +181,22 @@ def main():
...
+            module.fail_json(msg='Error while reading CRL file from disk: {0}'.format(e))
+    else:
+        data = module.params['content'].encode('utf-8')
+        if not identify_pem_format(data):
+            try:
+                data = base64.b64decode(module.params['content'])
+            except Exception as e:
```
Comment (Ajpantuso) [inline plugins/modules/x509_crl_info.py:195]:
```suggestion
            except (binascii.Error, TypeError) as e:
```
With `binascii` imported.

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
No functional issues as far as I can see and the added exception handling for `b64decode` triggered a proper module failure during my tests. Only comments are for explicit exceptions and the removal of some stale imports in the module script.

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Ajpantuso thanks a lot for reviewing this :)

**Pull request merged**